providers/standard/src/airflow/providers/standard/utils/skipmixin.py

```diff
@@ -0,0 +1,201 @@
+        tasks: Iterable[DAGNode],
+    ):
+        """
+        Set tasks instances to skipped from the same dag run.
+
+        If this instance has a `task_id` attribute, store the list of skipped task IDs to XCom
+        so that NotPreviouslySkippedDep knows these tasks should be skipped when they
+        are cleared.
+
+        :param ti: the task instance for which to set the tasks to skipped
+        :param tasks: tasks to skip (not task_ids)
+        """
+        # SkipMixin may not necessarily have a task_id attribute. Only store to XCom if one is available.
+        task_id: str | None = getattr(self, "task_id", None)
+        task_list = _ensure_tasks(tasks)
+        if not task_list:
+            return
+
+        task_ids_list = [d.task_id for d in task_list]
+
+        if task_id is not None:
+            ti.xcom_push(
+                key=XCOM_SKIPMIXIN_KEY,
+                value={XCOM_SKIPMIXIN_SKIPPED: task_ids_list},
+            )
+
+        self._set_state_to_skipped(task_ids_list, ti.map_index)
+
+    def skip_all_except(
+        self,
+        ti: RuntimeTaskInstanceProtocol,
+        branch_task_ids: None | str | Iterable[str],
+    ):
+        """
+        Implement the logic for a branching operator.
+
+        Given a single task ID or list of task IDs to follow, this skips all other tasks
+        immediately downstream of this operator.
+
+        branch_task_ids is stored to XCom so that NotPreviouslySkippedDep knows skipped tasks or
+        newly added tasks should be skipped when they are cleared.
+        """
+        # Ensure we don't serialize a generator object
+        if branch_task_ids and isinstance(branch_task_ids, GeneratorType):
+            branch_task_ids = list(branch_task_ids)
+        log = self.log  # Note: need to catch logger form instance, static logger breaks pytest
+        if isinstance(branch_task_ids, str):
+            branch_task_id_set = {branch_task_ids}
+        elif isinstance(branch_task_ids, Iterable):
+            branch_task_id_set = set(branch_task_ids)
+            invalid_task_ids_type = {
+                (bti, type(bti).__name__) for bti in branch_task_id_set if not isinstance(bti, str)
+            }
+            if invalid_task_ids_type:
+                raise AirflowException(
+                    f"'branch_task_ids' expected all task IDs are strings. "
+                    f"Invalid tasks found: {invalid_task_ids_type}."
+                )
+        elif branch_task_ids is None:
+            branch_task_id_set = set()
+        else:
+            raise AirflowException(
+                "'branch_task_ids' must be either None, a task ID, or an Iterable of IDs, "
+                f"but got {type(branch_task_ids).__name__!r}."
+            )
+
+        log.info("Following branch %s", branch_task_id_set)
+
+        if TYPE_CHECKING:
+            assert ti.task
+
+        task = ti.task
+        dag = TaskInstance.ensure_dag(ti)
+
+        valid_task_ids = set(dag.task_ids)
```

Comment:
Thanks to the new exception, I've realized that it tries to access the DB directrly from here.
Need some help with replacing it with Task SDK compatible version.

Reply:
Yeah `ensure_dag` is fine, we can ignore it for AF3 but more problematic is 

https://github.com/apache/airflow/blob/ed14473a599df5afa461450cb3625e7c7e53eb1d/providers/standard/src/airflow/providers/standard/utils/skipmixin.py#L179-L188